### PR TITLE
Register prefs in right places

### PR DESCRIPTION
**Pull request by darkdh**
1. Prefs used by keyed service must be registered or set default in
RegisterProfilePrefs of pref_service_builder_utils
2. RegisterProfilePrefs or RegisterLocalStatePrefs depends on scope

fix https://github.com/brave/brave-browser/issues/5414

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/chrome-browser-prefs-browser_prefs.cc.patch [hunk 1]
```diff
@@ -10,3 +10,11 @@ index 50312cdb0324e3dad18c77c96e5e609bfb259541..ed0ab71e9e08494abb5016bcb4e1fce6
  }
  
  // Register prefs applicable to all profiles.
+@@ -908,6 +909,7 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry,
+ #endif
+ 
+   RegisterProfilePrefsForMigration(registry);
```
Comment (bridiver) [inline patches/chrome-browser-prefs-browser_prefs.cc.patch]: do we need a patch here? Can't we just do a chromium_src override for `RegisterProfilePrefs`?

File: patches/chrome-browser-profiles-profile_impl.cc.patch [hunk 2]
```diff
@@ -10,13 +10,3 @@ index 71619ca3d5271b0f813110dce0e5570b6eebdda7..a9c1f8ec99aafa581c7fc8261e470261
  #include "build/build_config.h"
  #include "chrome/browser/background/background_contents_service_factory.h"
  #include "chrome/browser/background_fetch/background_fetch_delegate_factory.h"
-@@ -508,6 +509,9 @@ ProfileImpl::ProfileImpl(
-   LoadPrefsForNormalStartup(async_prefs);
- #endif
- 
-+// Done here instead of browser_prefs.cc so it can override default prefs
```
Comment (bridiver) [inline patches/chrome-browser-profiles-profile_impl.cc.patch]: @darkdh did you verify that this still works correctly?

Comment (darkdh) [reply patches/chrome-browser-profiles-profile_impl.cc.patch]: yes, if we override default at wrong place, we will get a DCHECK failure saying using unregistered prefs

File: patches/chrome-browser-prefs-browser_prefs.cc.patch (hunk 1 above)
Comment (darkdh) [reply patches/chrome-browser-prefs-browser_prefs.cc.patch]: that will also override the calling from `RegisterUserProfilePrefs` and `RegisterSigninProfilePrefs` to be `RegisterProfilePrefs_ChromiumImpl` and no one calls our overridden `RegisterProfilePrefs`

Comment (darkdh) [reply patches/chrome-browser-prefs-browser_prefs.cc.patch]: Also it overrides `extensions::ActivityLog::RegisterProfilePrefs` => `extensions::ActivityLog::RegisterProfilePrefs_ChromiumImpl` and bunch of same cases

**Comment by darkdh** [comment]
this only error https://staging.ci.brave.com/job/brave-browser-build-pr/job/reg-pref-rework/9/execution/node/359/log/ doesn't seem like related to code changes
cc @mihaiplesa

**Comment by darkdh** [comment]
CI passed based on https://staging.ci.brave.com/job/brave-browser-build-pr/job/reg-pref-rework/9/ and https://staging.ci.brave.com/job/brave-browser-build-pr/job/reg-pref-rework/10/

**Pull request merged**